<?xml version="1.0"?>
<launch>
  <!--arg name="uri1" default="radio://0/80/2M/0xE7E7E7E706" /-->
  <!--arg name="uri1" default="radio://0/80/2M/0xE7E7E7E7E7" />
  <arg name="uri2" default="radio://0/100/2M/0xE7E7E7E712" />
  <arg name="uri3" default="radio://0/120/2M/0xE7E7E7E708" /-->


  <!--arg name="uri1" default="radio://0/120/2M/0xE7E7E7E708" /-->
  <!--arg name="uri1" default="radio://0/120/2M/0xE7E7E7E703" /-->
  <arg name="uri1" default="radio://0/80/2M/0xE7E7E7E701" />
  <!--arg name="uri2" default="radio://0/100/2M/0xE7E7E7E712" /-->
  <arg name="uri2" default="radio://0/100/2M/0xE7E7E7E702" />
  <!--arg name="uri3" default="radio://0/80/2M/0xE7E7E7E706" /-->
  <arg name="uri3" default="radio://0/120/2M/0xE7E7E7E703" />

  <arg name="frame" default="/world" />
  <!--arg name="ip" default="192.168.123.12" /-->
  <!--arg name="ip" default="192.168.1.104" /-->
  <arg name="ip" default="192.168.1.2" />

  <arg name="port" default="22223" />

  

  <!--arg name="cf_names_arg" value="CF1,CF2,CF3" /-->
  <arg name="cf_names_arg" value="CF1,CF2,CF3" />
  <param name="cf_names" value="$(arg cf_names_arg)" />


  <!--arg name="cf_names_arg" value="CF1,CF2,CF3" /-->
  <arg name="cf_names_arg2" value="CF1,CF2" />
  <param name="cf_names2" value="$(arg cf_names_arg2)" />


  <arg name="cf_jobs_arg" value="2,1,3" />
  <!--arg name="cf_jobs_arg" value="2,2,1" /-->
  <param name="cf_jobs" value="$(arg cf_jobs_arg)" />


  <param name="work_width" value="4" />
  <param name="work_length" value="5" />

  <arg name="charge_type_arg" value="Static" />
  <param name="charge_type" value="$(arg charge_type_arg)" />


  <arg name="replan_arg" value="True"/>
  <param name="replan" value="$(arg replan_arg)" />

  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>

  <group ns="CF1" if="$(eval 'CF1' in cf_names_arg)">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri1)" />
      <param name="tf_prefix" value="CF1" />
      <param name="enable_logging_battery" value="True" />
    </node>
  </group>

  <group ns="CF2" if="$(eval 'CF2' in cf_names_arg)">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri2)" />
      <param name="tf_prefix" value="CF2" />
      <param name="enable_logging_battery" value="True" />
    </node>
  </group>

  <group ns="CF3" if="$(eval 'CF3' in cf_names_arg)">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri3)" />
      <param name="tf_prefix" value="CF3" />
      <param name="enable_logging_battery" value="True" />
    </node>
  </group>

    <!--remap from="/CF2/CF2/external_position" to="external_position" />
    <remap from="/CF2/CF2/lost_frames" to="lost_frames" /-->
    <node name="cf_qualisys_node" pkg="crazyflie_scripts" type="qualisys_cf_stream_hanger.py" output="screen">
      <param name="ip" value="$(arg ip)" />
    </node>

    <!--remap from="/CF2/CF2/cmd_stop" to="cmd_stop" />
    <remap from="/CF2/CF2/cmd_position" to="cmd_position" /-->
    <node name="run_crazyflies" pkg="crazyflie_scripts" type="run_crazyflies_hanger.py" output="screen">
    </node>

    <!--node name="run_detection" pkg="crazyflie_scripts" type="Detection_hanger.py" output="screen">
    </node-->


</launch>
